src/libraries/System.IO.Compression/src/System/IO/Compression/ZipBlocks.cs

```diff
@@ -670,6 +670,62 @@ public static bool TrySkipBlock(Stream stream)
             bytesRead = stream.ReadAtLeast(blockBytes, blockBytes.Length, throwOnEndOfStream: false);
             return TrySkipBlockFinalize(stream, blockBytes, bytesRead);
         }
+
+        public static bool TrySkipBlockAESAware(Stream stream, out byte? aesStrength, out ushort? originalCompressionMethod, out ushort? aesVersion, out uint? crc32)
+        {
+            aesStrength = null;
+            originalCompressionMethod = null;
+            aesVersion = null;
+            crc32 = null;
+            BinaryReader reader = new BinaryReader(stream);
+
+            // Read the first 4 bytes (local file header signature)
+            byte[] signatureBytes = reader.ReadBytes(4);
+            if (!signatureBytes.AsSpan().SequenceEqual(ZipLocalFileHeader.SignatureConstantBytes))
+            {
+                return false; // Not a valid local file header
+            }
+
+            // Read fixed-size fields after signature
+            // Local file header layout:
+            // signature (4) + version (2) + flags (2) + compression (2) +
+            // mod time (2) + mod date (2) + CRC32 (4) + compressed size (4) +
+            // uncompressed size (4) + name length (2) + extra length (2)
+
+            reader.ReadBytes(10); // Skip version through mod date
+            crc32 = reader.ReadUInt32(); // Read CRC32
+            reader.ReadBytes(8); // Skip compressed and uncompressed sizes
+            ushort nameLength = reader.ReadUInt16();
+            ushort extraLength = reader.ReadUInt16();
+
+            // Skip file name
+            stream.Seek(nameLength, SeekOrigin.Current);
+
+            // Parse extra fields
+            long extraStart = stream.Position;
+            long extraEnd = extraStart + extraLength;
+            while (stream.Position < extraEnd)
+            {
+                ushort headerId = reader.ReadUInt16();
+                ushort dataSize = reader.ReadUInt16();
+
+                if (headerId == 0x9901) // AES extra field
```

Comment:
```suggestion
                if (headerId == WinZipAesExtraField.HeaderId)
```

Maybe also consider moving reading logic to WinZipAesExtraField type, since it already contains write logic